Scrum-120 - Class Code View Page FE

weswilson24 opened the pull request:
Here is the frontend only page for the Class code view. I will connect the backend for the actually data rows later this week

1. Shows all available classes.
2. Can go through multiple pages of classes with the arrows with maximum of 5 rows per page
3. Can type in letters or class name in the search bar to see if a specific class exists
![Screenshot 2024-09-23 131838](https://github.com/user-attachments/assets/79b27505-e0ef-465b-86bf-d990bdf0400c)
![Screenshot 2024-09-23 131817](https://github.com/user-attachments/assets/ab35151e-22b8-4732-ab57-a767a6971781)
![Screenshot 2024-09-23 131755](https://github.com/user-attachments/assets/79d017d7-295e-4f9e-a27c-e8743151f394)

mengzhuou commented:
1. Please make the scroll bar invisible
<img width="1008" alt="image" src="https://github.com/user-attachments/assets/bbbd3d7e-2d13-4107-8c8c-af9ed23b8eef">

2. Please connect the "Back" button with Select task page, if it does not exist, connect with an existing page.

3. Color of Search button is not synced with Figma
<img width="319" alt="image" src="https://github.com/user-attachments/assets/4a48fb5c-0b0f-4a1f-bf5c-95ad4faad95b">

Xisheng0725 reviewed (changes requested):
the class data should using the backend api since we already have those api, you can use postman to create and post some data in order to get and use.

mengzhuou commented on frontend/src/components/Pages/ClassCodesAdmin/ClassCodesAdmin.js:
```diff
@@ -0,0 +1,126 @@
+
+   
+    const handleEditClass = () => {
+        navigate("/ClassCodesEdit", { state: { classData } });
```
maybe we want to use existing withFuncProps function for all navigation.

weswilson24 commented:
New Changes:
1. Got ride of sidebar
2. Changed faunt and button to match those of previous pages
3. Changed color of search image icon
4. Connected to database data and search functions is fully functionable with this
5. Connecting it to ClassCodesEdit page in SCRUM 121
6. Currently goes to ViewEquipment when the back button is hit
7. Not all data has all four categories per column
8. Edited backend code to allow for a deleteClassCode, addClassRow and update classRow for use on EditClassCodePage
9. Only added getClassCodes to connector because that is all that is need for this page 
10. Fixed pagination arrows 
![Screenshot 2024-09-25 171305](https://github.com/user-attachments/assets/d60f5256-257a-4bc2-8223-1afa623898ef)

Xisheng0725 reviewed (changes requested):
the package name is not attribute for class code, it's in the equipment api for the bundle items, the package name should be the bundle name in equipment api. It's a bit confusing but please check below figma design:
<img width="565" alt="image" src="https://github.com/user-attachments/assets/e9d5df1e-e9cf-46e6-a910-bdcdf9b14729">

Also,if profeesor didn't choose to add package in the code, please just show N/A at there

Qihuiwang3 reviewed (changes requested):
When I click the edit button, it takes me to the enter class code page.
Also when user hover on the icon, the cursor should become pointer and change the dark blue #003a99
<img width="37" alt="image" src="https://github.com/user-attachments/assets/64923f81-c593-41ea-835b-13dc3cc1db04">

Qihuiwang3 reviewed (changes requested):
All the functionalities I tested are working correctly. The only thing that needs to be updated is the UI. Since this page is similar to the “Students” page (both have edit and non-edit modes), which Nina has already completed and merged into the main branch, I suggest you refer to her implementation for consistency.
You can find it here: http://localhost:3000/Students

**Summary of some UI inconsistencies I noticed:**
	1. When hovering over each row, the row should be highlighted in blue.
	2. The background of the table shouldn’t be white.
	3. The pagination style looks different.
	4. The search icon appears different (it seems lighter). You can use the one Nina implemented on the Students page.
	5.  The 'Add New'modal is different 

There might be additional issues, so it would be great if Nina @mengzhuou could review it as well.

**Also, something important:**
I updated the Figma this past weekend to improve the overall logic and user experience. The change is under the edit mode: when you click the edit icon, a modal should pop up allowing users to update any information. It’s the exact same modal as the “Add New” modal you’ve already created, with the only difference being that this edit modal should display existing information. You can refer to the attached screenshot for reference.

I understand this is an additional change that wasn’t part of the original ticket, but it would be very helpful if you could finish this in this sprint. Since we already have a lot to finish for this project within one semester, completing this now would prevent pushing it to the next sprint and greatly help our efficiency in finishing the project. We’re also not certain about any future difficulties that may arise, so getting this done sooner would be ideal.

I apologize for the late change, and I greatly appreciate your understanding. Please let me know if you’re able to complete this in this sprint, it would be very awesome. If not, it's also understandable. Again, I am so sorry for the late change. 
 
 **Here is the Figma screenshot for your reference:** 
<img width="831" alt="image" src="https://github.com/user-attachments/assets/5c8670cc-858e-4a77-bc15-a50304fc397f">

**Here is the Figma Link, just in case you need it.** https://www.figma.com/design/oNrdhWZGmiPs1XSKcJXHpa/LMC-Film-Inventory-Check-out-System?node-id=375-97&node-type=canvas&t=qr1k5ewJTR2KiGNS-0

weswilson24 commented:
1. All functionality basically done. 
2. 
![Screenshot 2024-09-26 231104](https://github.com/user-attachments/assets/2d6f9670-5035-4ee0-812c-764b04577176)
![Screenshot 2024-09-26 231033](https://github.com/user-attachments/assets/800d2a62-9025-46ec-8245-80036bc45a52)
![Screenshot 2024-09-26 231018](https://github.com/user-attachments/assets/29b22439-6bf1-4517-ad24-6dc89866d0ad)
![Screenshot 2024-09-26 230959](https://github.com/user-attachments/assets/fd6c21d6-5a08-4816-b8d8-cbefe96094fa)
3.Ignore battery saver on bottom right you hate to see it

Xisheng0725 reviewed (changes requested):
Thank you. I think this version looks good, but in connector.js I think you missed the "," during fixed merge conflict which causing thr run time error. since we already have prev page of admin class code page, please connect with prev page based on figma design. Also I think your back button connect the wrong page, it should connect to the management page of class code button. After you fixed those, I think it's good to merge.

Xisheng0725 reviewed (changes requested):
the back button should link to /Management page instead of select task page and after click the class code and press confirm, it should go to your page

mengzhuou commented:
1. The columns are not consistent when going to next page as it shift from left to right
2. The input field should not display [Object, Object]
<img width="611" alt="image" src="https://github.com/user-attachments/assets/40e8d9c5-9aeb-4b8a-8a54-f81ba6ea5c66">

3. Buttons are not sticky to the bottom when there's less row size
<img width="902" alt="image" src="https://github.com/user-attachments/assets/e1a97fcd-d02f-4a89-88f6-5fea7da04dda">
